### DISCUSSION TITLE
php-cs-fixer integration

### DISCUSSION THREAD

**Discussion by ziriuz84** [LazyVim/LazyVim#2035] [Q&A]
Hi all! I installed php-cs-fixer via Mason. How can I use it as my standard formatter?

Thanks a lot

**Comment by dimas-cyriaco** [comment]
LazyVim uses [Conform](https://www.lazyvim.org/plugins/formatting#conformnvim) for formatting.

Here's how I set it up:

```lua
return {
  "stevearc/conform.nvim",
  opts = {
    formatters_by_ft = {
      sql = { "sql_formatter" },
    },
  },
}
```

**Comment by ziriuz84** [comment]
Does It creates a completely new config or does it update the default one?

**Reply by dpetka2001** [reply]
You should really read the [docs](https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-customizing-plugin-specs) to answer questions like that. Reading `lazy.nvim`'s docs would also go a long way for a deeper understanding if you'd like to know more.

**Reply by ziriuz84** [reply]
I'm using this setup, but it doesn't run php-cs-fixer, it's formatting as if it uses prettier. If I run php-cs-fixer manually on the file it formats well, with the right indentation

```lua
return {
  {
    "stevearc/conform.nvim",
    opts = function()
      local opts = {
        formatters_by_ft = {
          python = { "autoflake", "isort", "black", "autopep8" },
          php = { "php-cs-fixer" },
          go = { "gofmt", "goimports", "gomodifytags", "djlint" },
          rust = { "rustfmt" },
          javascript = { "prettier", "rustywind" },
          typescript = { "prettier", "rustywind" },
          JSON = { "fixjson", "prettier" },
          sh = { "beautysh", "shfmt" },
          blade = { "php-cs-fixer", "blade-formatter" },
          YAML = { "prettier", "yamlfmt", "yamlfix" },
        },
      }
      return opts
    end,
  },
}
```

**Reply by madflow** [reply]
I think it should be `php_cs_fixer`. 

`:help conform-formatters`

**Reply by ziriuz84** [reply]
Yes, now It works. It's only a little slow on big files and It goes on timeout very often. I'll search ho how ti set a timeout
